contributing/code/conventions.rst

```diff
@@ -79,31 +79,47 @@ must be used instead (where ``XXX`` is the name of the related thing):
 
 .. _contributing-code-conventions-deprecations:
 
-Deprecations
-------------
+Deprecating code
+----------------
 
 From time to time, some classes and/or methods are deprecated in the
 framework; that happens when a feature implementation cannot be changed
 because of backward compatibility issues, but we still want to propose a
 "better" alternative. In that case, the old implementation can simply be
 **deprecated**.
 
+Deprecations must only be introduced on the next minor version of the impacted
+component (or bundle, bridge, contract).
+They can exceptionally be introduced on previous supported versions if they are critical.
+
+A new class (or interface, trait) cannot be introduced as deprecated, or
+contains deprecated methods.
```

Comment:
`contains` -> `contain`  (because it's the continuation of the phrase, so `cannot ... contain`)